analyses/tmb-vaf-longitudinal/util/function-create-barplot.R

```diff
@@ -0,0 +1,71 @@
+#' Create stacked barplots for all samples in the df
+#'
+#' @param tmb_df_df 
+#' @param ylim 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+create_stacked_barplot <- function(tmb_df, ylim) {
+
+  # Rename legend for timepoints
+  Timepoint <- factor(tmb_df$tumor_descriptor)
+  
+  # Define and order palette
+  palette <- palette_df$hex_codes
```

Comment:
I am not 100% sure but in my understanding, `palette_df` is an argument here (with the same value that is sent to the function when it is called). Parameters are values enclosed in parentheses whenever a function is created. These are subject to change according to what one passes in the function. Does this make sense?
@jharenza what do you say?

Reply:
I think what ryan is trying to say is that a user may run this function but not have `palette_df` loaded, and then this would fail, so input of the df should be a required part of the function if used within the function, if that makes sense?

Reply:
and, to that end, does the palette_df need special formatting (eg - are certain column names required, or is this agnostic)

Reply:
There is no formatting for `palette_df`. In the function, I use one column for the values (specific colors) and the other column to retrieve the names.

```
# Define and order palette
  palette <- palette_df$hex_codes
  names(palette) <- palette_df$color_names
```

So, does this mean that it is preferable to be part of the parameters?

Reply:
Yes because you are setting a requirement above that only the `tmb_df` and `ylim` are requirements to run the function. If `palette_df` is optional, but still desired, you'd include it but then you'd want to make an ifelse whether it is input such that if someone does not have a `palette_df` with required colnames, then you would plot colors differently. Does that make sense? Without a requirement, the assumption here is that someone who uses this function will have `palette_df`, loaded.

Reply:
Okay, thanks for the clarifirication!

Reply:
This has been addressed now. See ca7f850be8c68ede2b748b3fc2c850ed7aada46e and 1aa1fe83ae2bd5b39a44b303e40ca4003ddcb3a2